lib/instrumenter/collectors-vm/link.js

```diff
@@ -45,13 +46,64 @@ var LINK_KSTATS = [
     }
 ];
 
+/*
+ * Because this is a latency-sensitive code path, we can't pay the
+ * price of forking a new process to run `vmadm get <:instance_uuid>`.
+ * Instead, we invoke vminfod local HTTP API --  which has not been
+ * stabilized yet -- to retrieve the instance information.
+ */
+
+var VMINFOD_TIMEOUT_MS = 500;
+
+function getInstanceNicsInfo(instanceUuid, cb) {
+
+    var timeout = {};
+    function done(err, data) {
+        if (!timeout.done) {
+            cb(err, data);
+            timeout.done = true;
+        }
+    }
```

Comment:
It seems to me like this will swallow the error and log nothing if `timeout.done === true`.  Would it be better to still log something if there is an error and `timeout.done === true` or is that state not even possible?

Reply:
If `timedout === true`, then we already have an http response. We don't need to call `cb` in this case.